File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc():
    return [i for i in range(a, b)] #Create list of ints from a to b, Haskell equivalent [a..b]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/(b-a+1)

print(applicatorFunc(listFunc, 'a', 1, 10))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc a b = [a..b] 

--Define applicatorFunc
applicatorFunc inpFunc a b s = if s=='s' then sum (inpFunc a b) else (sum (inpFunc a b))/(b-a+1)

--tried to write using guards
--applicatorFunc a b s | s == 's' = sum (inpFunc a b)
--                     | otherwise = (sum(inpFunc a b))/(b-a+1)


main = do
    let result = applicatorFunc inpFunc 1 10 's' --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show result)